pennylane/operation.py

```diff
@@ -964,6 +964,75 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
 
+class Projection(Operation):
+    """Base class for projection operations
+
+    As with :class:`~.Operation`, the following class attributes must be
+    defined for all projections:
+
+    * :attr:`~.Operator.num_params`
+    * :attr:`~.Operator.num_wires`
+    * :attr:`~.Operator.par_domain`
+
+    To define a projection, the following attribute of :class:`~.Projection`
+    can be used to list the corresponding projectors.
+
+    * :attr:`~.Projection._projectors`
+
+    Args:
+        params (tuple[float, int, array, Variable]): operation parameters
+
+    Keyword Args:
+        wires (Sequence[int]): Subsystems the projection acts on. If not given, args[-1]
+            is interpreted as wires.
+        do_queue (bool): Indicates whether the operation should be
```

Comment:
Does `do_queue` also appear in other operations? Seems new to me, but I guess I missed those changes

Reply:
I believe it does, but I'll check again!

Reply:
@josh146 does `do_queue` still apply to the new QNode/tape core?

Reply:
>does do_queue still apply to the new QNode/tape core?

No it doesn't :)

Reply:
In that case, should this be removed @josh146? I was just trying to keep it consistent with the other classes.

Reply:
Keep it in for now; when it comes time to rip out the old pennylane core, this can be removed en masse